src/Microsoft.DotNet.ProjectModel/Files/IncludeContext.cs

```diff
@@ -0,0 +1,175 @@
+                    if (defaultBuiltInExclude != null && !BuiltInsExclude.Any())
+                    {
+                        BuiltInsExclude = defaultBuiltInExclude.ToList();
+                    }
+                }
+                var mappings = token.Value<JToken>("mappings") as JObject;
+                if (mappings != null)
+                {
+                    Mappings = new List<KeyValuePair<string, IncludeContext>>();
+                    foreach (var map in mappings)
+                    {
+                        Mappings.Add(new KeyValuePair<string, IncludeContext>(map.Key, new IncludeContext(sourceBasePath, map.Key, mappings)));
+                    }
+                }
+            }
+        }
+
+        public string SourceBasePath { get; }
+
+        public string Option { get; }
+
+        public List<string> IncludePatterns { get; }
+
+        public List<string> ExcludePatterns { get; }
+
+        public List<string> IncludeFiles { get; }
+
+        public List<string> ExcludeFiles { get; }
+
+        public List<string> BuiltInsInclude { get; }
+
+        public List<string> BuiltInsExclude { get; }
+
+        public List<KeyValuePair<string, IncludeContext>> Mappings { get; }
+
+        public override bool Equals(object obj)
+        {
+            var other = obj as IncludeContext;
+            return other != null &&
+                SourceBasePath == other.SourceBasePath &&
+                Option == other.Option &&
+                EnumerableEquals(IncludePatterns, other.IncludePatterns) &&
+                EnumerableEquals(ExcludePatterns, other.ExcludePatterns) &&
+                EnumerableEquals(IncludeFiles, other.IncludeFiles) &&
+                EnumerableEquals(ExcludeFiles, other.ExcludeFiles) &&
+                EnumerableEquals(BuiltInsInclude, other.BuiltInsInclude) &&
+                EnumerableEquals(BuiltInsExclude, other.BuiltInsExclude) &&
+                EnumerableEquals(Mappings, other.Mappings);
+        }
+
+        public override int GetHashCode()
+        {
+            return base.GetHashCode();
+        }
+
+        private static bool EnumerableEquals<T>(IEnumerable<T> left, IEnumerable<T> right)
+            => Enumerable.SequenceEqual(left ?? EmptyArray<T>.Value, right ?? EmptyArray<T>.Value);
+
+        private static string[] ExtractValues(JToken token)
+        {
+            if (token != null)
+            {
+                if (token.Type == JTokenType.String)
+                {
+                    return new string[] { token.Value<string>() };
+                }
+
+                else if (token.Type == JTokenType.Array)
+                {
+                    return token.Values<string>().ToArray();
+                }
+            }
+
+            return new string[0];
+        }
+
+        private static List<string> CreateCollection(string projectDirectory, string propertyName, IEnumerable<string> patternsStrings, bool literalPath)
```

Comment:
too long, wrap this line. one golden rule is that the line should be short enough to fix in github review.